packages/fxa-auth-server/test/scripts/stripe-products-and-plans-converter.js

```diff
@@ -0,0 +1,796 @@
+    };
+    const product2 = deepCopy({ ...product1, id: 'prod_456' });
+    const productConfig2 = deepCopy({
+      ...productConfig1,
+      stripeProductId: product2.id,
+    });
+    const plan1 = deepCopy({
+      ...plan,
+      metadata: {
+        'capabilities:aFakeClientId12345': 'more, comma, separated,  values',
+        upgradeCTA: 'hello <a href="http://example.org">world</a>',
+        productOrder: '2',
+        downloadURL: 'https://example.com/download',
+      },
+      id: 'plan_123',
+    });
+    const planConfig1 = {
+      active: true,
+      stripePriceId: plan1.id,
+      capabilities: {
+        aFakeClientId12345: ['more', 'comma', 'separated', 'values'],
+      },
+      uiContent: {
+        upgradeCTA: 'hello <a href="http://example.org">world</a>',
+      },
+      urls: {
+        download: 'https://example.com/download',
+      },
+      productOrder: 2,
+    };
+    const plan2 = deepCopy({ ...plan1, id: 'plan_456' });
+    const planConfig2 = { ...deepCopy(planConfig1), stripePriceId: plan2.id };
+    const plan3 = deepCopy({ ...deepCopy(plan1), id: 'plan_789' });
+    const planConfig3 = { ...deepCopy(planConfig1), stripePriceId: plan3.id };
+    beforeEach(() => {
+      const firestore = setupFirestore(mockConfig);
+      Container.set(AuthFirestore, firestore);
+      Container.set(AuthLogger, {});
+      Container.set(AppConfig, mockConfig);
+      paymentConfigManager = new PaymentConfigManager();
+      Container.set(PaymentConfigManager, paymentConfigManager);
+      productConfigDbRef = paymentConfigManager.productConfigDbRef;
+      planConfigDbRef = paymentConfigManager.planConfigDbRef;
+      converter = new StripeProductsAndPlansConverter({
+        log: mockLog,
+        stripeHelper: mockStripeHelper,
+        supportedLanguages: mockSupportedLanguages,
+      });
+      args = {
+        productId: '',
+        isDryRun: false,
+      };
+      async function* productGenerator() {
+        yield product1;
+        yield product2;
+      }
+      async function* planGenerator1() {
+        yield plan1;
+        yield plan2;
+      }
+      async function* planGenerator2() {
+        yield plan3;
+      }
+      converter.stripeHelper.stripe = {
+        products: { list: sandbox.stub().returns(productGenerator()) },
+        plans: {
+          list: sandbox
+            .stub()
+            .onFirstCall()
+            .returns(planGenerator1())
+            .onSecondCall()
+            .returns(planGenerator2()),
+        },
+      };
+    });
+    afterEach(async () => {
+      await deleteCollection(
+        paymentConfigManager.firestore,
+        productConfigDbRef,
+        100
+      );
+      await deleteCollection(
+        paymentConfigManager.firestore,
+        planConfigDbRef,
+        100
+      );
+      Container.reset();
+      sandbox.reset();
+    });
+    it('processes new products and plans', async () => {
+      await converter.convert(args);
+      await retry(
```

Comment:
Was this retry for the race condition on the cached list you described above?  Do we still need it?

Reply:
Yes thank you I can definitely remove that now!